### PR TITLE
Sub timeout - remove unused F wrappers.

### PR DESCRIPTION
**Pull request by diesalbla**
The external `F.delay` wrappers was not useful in any way, since they were being immediately unwrapped in the body of `task`, so they have no effect but cost allocations.

Note that it is the function itself that delays the side effectful operations on `dom.window` in acquire.

### CODE REVIEW

File: tyrian/js/src/main/scala/tyrian/Sub.scala [hunk 1]
```diff
@@ -175,28 +175,20 @@ object Sub:
...
   /** A subscription that produces a `msg` after a `duration`. */
   def timeout[F[_]: Sync, Msg](duration: FiniteDuration, msg: Msg, id: String): Sub[F, Msg] =
-    val acquire: F[(Either[Throwable, Msg] => Unit) => Int] =
-      Sync[F].delay { (callback: Either[Throwable, Msg] => Unit) =>
+    val acquire: (Either[Throwable, Msg] => Unit) => Int =
+      (callback: Either[Throwable, Msg] => Unit) =>
         dom.window.setTimeout(
           Functions.fun0(() => callback(Right(msg))),
           duration.toMillis.toDouble
         )
```
Comment (davesmith00000) [inline tyrian/js/src/main/scala/tyrian/Sub.scala:183]: I'll take another look at this in the morning when I'm a bit fresher, but my only slight concern here would be that we lose error handling. In practice, I don't think there are any errors to handle though... :thinking:

**Review by davesmith00000 — COMMENTED** [review]
Thanks @diesalbla! I've kicked off a build, looks a lot cleaner. :+1:

File: tyrian/js/src/main/scala/tyrian/Sub.scala (hunk 1 above)
Comment (diesalbla) [reply tyrian/js/src/main/scala/tyrian/Sub.scala:183]: So, the `Sync[F].delay` covers the creation of the function, but it does not cover an execution of the body of the function.

**Comment by davesmith00000** [comment]
Good morning @diesalbla! Looks like a CI fail, how are you building Tyrian locally?

**Review by davesmith00000 — APPROVED** [review]
LGTM.

I had to think about that a bit. I expect it works because `handle` is eagerly evaluated and then the cancellation is delayed. It's definitely more efficient, however I'm not sure in general about the cost of the mental gymnastics of flipping mode from greedy to lazy Scala though.

In small cases like this where there are obviously (I hope!) no issues, it's probably ok. For anything much bigger I'd probably rather it was one or the other, i.e. CE or Vanilla / Lazy or Greedy.

**Pull request merged**